### DISCUSSION TITLE
Set cookies for request via 'httpx.Cookies()' does not work

### DISCUSSION THREAD

**Discussion by simon-liebehenschel** [encode/httpx#2179] [General]
### Reproducible code sample:
```python
cookies = [
    {
        "name": "foo",
        "value": "bar",
        "domain": "exampl.com",
        "path": "/",
        "expires": 1681290623,
        "httpOnly": True,
        "secure": True,
        "sameSite": "None",
    },
    {
        "name": "spam",
        "value": "eggs",
        "domain": "exampl.com",
        "path": "/",
        "expires": 1681290623,
        "httpOnly": True,
        "secure": True,
        "sameSite": "None",
    },
]
container = httpx.Cookies()
for cookie in cookies:
    container.set(name=cookie["name"], value=cookie["value"], domain=cookie["domain"], path=cookie["path"])
r = httpx.get(
    "https://httpbin.org/cookies/set",
    cookies=container,
    follow_redirects=True,
)
print(r.text)
```
Response output:
```
{
  "cookies": {}
}
```

### Versions
- httpx  0.22.0
- Python 3.10

**Comment by simon-liebehenschel** [comment]
Here is an issue that was closed without any explanations https://github.com/encode/httpx/issues/2178

**Reply by lovelydinosaur** [reply]
> that was closed without any explanations

Nope, it was moved to a discussion.

(Which our contribution guidelines are super clear about)

![Screen Shot 2022-04-15 at 19 46 37](https://user-images.githubusercontent.com/647359/163615231-0b762060-f16a-4510-aa68-c55889aa1e36.png)

**Reply by simon-liebehenschel** [reply]
@tomchristie I am sorry for my unawareness. Many open source repositories even do not have 'discussions' enabled, so I often open an issue right away for possible bugs. Thank you for letting me know! I will follow 'httpx' specific requirements further.
